src/AstSemanticChecker.cpp

```diff
@@ -430,11 +430,35 @@ void AstSemanticChecker::checkLiteral(
         }
     }
 }
+/**
+ * agg1, agg2 are clauses which contain no head, and consist of a single literal
+ * that contains an aggregate.
+ * agg1 is dependent on agg2 if agg1 contains a variable which is grounded by agg2, and not by agg1.
+ */
+bool AstSemanticChecker::isDependent(const AstClause& agg1, const AstClause& agg2) {
+    auto grounded1 = getGroundedTerms(agg1);
+    auto grounded2 = getGroundedTerms(agg2);
+    bool dependent = false;
+    visitDepthFirst(agg1, [&](const AstVariable& var) {
+        const AstVariable* var2 = nullptr;
+        visitDepthFirst(agg2, [&](const AstVariable& v2) {
+            if (v2.getName() == var.getName()) {
+                var2 = &v2;
+                return;
+            }
+        });
+        if (var2 != nullptr) {
+            if (!grounded1[&var] && grounded2[var2]) dependent = true;
```

Comment:
```suggestion
            if (!grounded1[&var] && grounded2[var2]) {
                dependent = true;
            }
```
That aside, it might be better to do
```
if (var2 == nullptr) {
    return false;
}
return !grounded1[&var] && grounded2[var2];

Reply:
I would like that but my problem was i'm inside the lambda function so the return statement won't get me out of the isDependent function